subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap.c

```diff
@@ -160,14 +160,49 @@ int nrf_cloud_coap_pgps_url_get(struct nrf_cloud_rest_pgps_request const *const
 }
 #endif /* CONFIG_NRF_CLOUD_PGPS */
 
+int nrf_cloud_coap_obj_send(struct nrf_cloud_obj *const obj)
+{
+	if (!nrf_cloud_coap_is_connected()) {
+		return -EACCES;
+	}
+
+	if (!obj) {
+		return -EINVAL;
+	}
+
```

Comment:
i think we should add the following check, at least for the type... up to you if you want to allow a pre-encoded buffer to be sent (i would say no, to reduce potential errors/misuse):
```
/* Only support sending of the CoAP CBOR type or a pre-encoded buffer. */
if ((obj->type != NRF_CLOUD_OBJ_TYPE_COAP_CBOR) &&
    (obj->enc_src != NRF_CLOUD_ENC_SRC_PRE_ENCODED) {
		return -ENOTSUP;
}
```